src/usecases/search.ts

```diff
@@ -0,0 +1,14 @@
+import { type SearchDetailInput } from 'entities';
+import { type ISearchDataGateway } from './interfaces';
+import { validateData } from 'utils/helpers';
+import { searchInputSchema } from 'schemas/search';
+
+export class SearchUsecase {
+  constructor(private readonly dataGateway: ISearchDataGateway) {}
```

Comment:
This usecase is where all the logic of the search should be. 

What this means is that we add `eventsDatagateway`, `newsDatagateway` etc.

Then we add the logic of the search based on the results we get from the gateways. 

That way one can understand the logic from a high level